modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/function/MBFunction_match.java

```diff
@@ -0,0 +1,102 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ * 
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ * 
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ * 
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.mbstyle.function;
+
+import static org.geotools.filter.capability.FunctionNameImpl.parameter;
+
+import org.geotools.filter.FunctionExpressionImpl;
+import org.geotools.filter.capability.FunctionNameImpl;
+import org.geotools.mbstyle.parse.MBFormatException;
+import org.json.simple.JSONArray;
+import org.opengis.filter.capability.FunctionName;
+import org.opengis.filter.expression.Expression;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Selects the output whose label value matches the input value, or the fallback value if no match is found.
+ * The input can be any string or number expression (e.g. ["get", "building_type"]).
+ * Each label can either be a single literal value or an array of values.
+ * Example: ["match",
+ * input: InputType (number or string),
+ * label_1: InputType | [InputType, InputType, ...], output_1: OutputType,
+ * label_n: InputType | [InputType, InputType, ...], output_n: OutputType,
+ * ..., default: OutputType]: OutputType
```

Comment:
Please provide a proper CQL or code example as well (i.e. how you would call it when not using MBStyle), to better illustrate how it should be used within GeoTools)

How this function actually works is a bit unclear from a first glance, so a more concrete example, with actual sample values would also be helpful